### PR TITLE
[Improvement][Dependent] The project list in dependent node's permissions should not be restricted. 

### LINKED ISSUE #11902: [Improvement][Dependent] The project list in dependent node's permissions should not be restricted.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The project list in dependent node's permissions should not be restricted. Suggesting create a new API without project permission restrictions and provide it to dependent nodes in UI.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by limaiwang**
## Purpose of the pull request

close https://github.com/apache/dolphinscheduler/issues/11902
## Brief change log

The project list in dependent node's permissions should not be restricted. Suggesting create a new API without project permission restrictions and provide it to dependent nodes in UI.

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by limaiwang** [comment]
Hi @SbloodyS  Help to review whether the Issue requirements are met

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml [hunk 1]
```diff
@@ -180,4 +180,10 @@
             union select id as project_id  from t_ds_project where user_id=#{userId})
         </if>
     </select>
+
+    <select id="queryAllProjectForDependent" resultType="org.apache.dolphinscheduler.dao.entity.Project">
+        select
+        <include refid="baseSql"/>
+        from t_ds_project
```
Comment (SbloodyS) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml]: It is recommended to query only the required fields, such as name and code.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 2]
```diff
@@ -288,4 +288,23 @@ public ProjectListResponse queryAllProjectList(@ApiIgnore @RequestAttribute(valu
...
+     */
+    @ApiOperation(value = "queryAllProjectListForDependent", notes = "QUERY_ALL_PROJECT_LIST_FOR_DEPENDENT_NOTES")
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "loginUser", value = "LOGIN_USER", dataTypeClass = Object.class, example = "\"{id:100}\"", required = true)
+    })
+    @GetMapping(value = "/list-dependent")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiException(LOGIN_USER_QUERY_PROJECT_LIST_PAGING_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result queryAllProjectListForDependent(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser) {
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]:
```suggestion
    public ProjectListResponse queryAllProjectListForDependent(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser) {
```

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml (hunk 1 above)
Comment (limaiwang) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml]: fixed dependent node only gets item code and name, confirmed ui only gets code and name

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java (hunk 2 above)
Comment (limaiwang) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: Bug fixed

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java [hunk 3]
```diff
@@ -288,4 +288,23 @@ public ProjectListResponse queryAllProjectList(@ApiIgnore @RequestAttribute(valu
...
+    /**
+     * query all project list for dependent
+     *
+     * @param loginUser login user
+     * @return all project list
+     */
+    @ApiOperation(value = "queryAllProjectListForDependent", notes = "QUERY_ALL_PROJECT_LIST_FOR_DEPENDENT_NOTES")
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "loginUser", value = "LOGIN_USER", dataTypeClass = Object.class, example = "\"{id:100}\"", required = true)
+    })
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: It's useless. Please remove it.

Comment (limaiwang) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProjectV2Controller.java]: removed

**Review by SbloodyS — COMMENTED** [review]
Please resolve the conflicts.

**Pull request closed without merging**